src/App.js

```diff
@@ -1,16 +1,78 @@
 import React from 'react';
+import './components/TodoComponents/Todo.css';
+
+let collectCompleted = [];
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor(){
+    super();
+    this.state = {
+      toDoList: [],
+      task: "",
+      id: "",
+      completed: "",
+    };
+  }
+
+  toDoListHandler = event => {
+    this.setState({[event.target.name]: event.target.value});
+    
+  };
+
+  toDoListSubmit = event => {
+    event.preventDefault();
+    let toDo = {
+      task: this.state.task,
+      id: Date.now(),
+      completed: false,
+    };
+    this.setState(prevState => {
+      return{
+        toDoList: [...prevState.toDoList, toDo],
+        task: "",
+        id: "",
+        completed: false,
+        
+      };      
+    });
+    
+  }
+  clearCompleted = event =>{
+    event.preventDefault();
+    console.log(this.state.toDoList)
+    console.log(collectCompleted)
+    this.setState(prevState => {
+      return(
+        this.state.toDoList.filter((item) => {
+          !collectCompleted.includes(item.id) ? console.log('not completed') : item.completed = !false;
+        })
+      )
+    })
+  }
+  
+  
   render() {
     return (
       <div>
-        <h2>Welcome to your Todo App!</h2>
+        <h2>The ToDo App</h2>     
```

Comment:
you could have made this forum in it's own component and then imported the form here. 
In React you want to try and keep a minimum amount of code and logic in the app.js file when possible so everything is kept in components and is thus cleaner and more organized. Can help with debugging too.